src/Agent.Listener/Configuration/ServiceControlManager.cs

```diff
@@ -35,7 +35,19 @@ public void CalculateServiceName(AgentSettings settings, string serviceNamePatte
             serviceName = string.Empty;
             serviceDisplayName = string.Empty;
 
-            string accountName = new Uri(settings.ServerUrl).Host.Split('.').FirstOrDefault();
+            Uri accountUri = new Uri(settings.ServerUrl);
+            string accountName = string.Empty;
+
+            if (accountUri.Host == "dev.azure.com")
+            {
+                accountName = accountUri.AbsolutePath.Split("/").Skip(1).FirstOrDefault();
```

Comment:
maybe `AbsolutePath.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault()`?

Reply:
Sure, I'll address this later today.